### PR TITLE
FIX bug #1961

### LINKED ISSUE #1961: [BUG] in POST v2/subscriptions if servicePath header is empty, the servicePath field is missing in DB
in POST v2/subscriptions if Fiware-ServicePath header is empty, the servicePath field is missing in DB, when it should be "/" value as default.

#### Subscription request
```
POST http://qa-orion-fe-02:1026/v2/subscriptions
Content-Type: application/json
Fiware-Service: test_service_path
Fiware-ServicePath:
 {"notification": {"callback": "http://localhost:1234", "attributes": ["temperture_0"]}, "expires": "2016-04-05T14:00:00.00Z", "subject": {"entities": [{"idPattern": ".*", "type": "room"}], "condition": {"attributes": ["temperature_0"]}}}
```

#### Subscription response
```
http code: 201
date: Tue, 29 Mar 2016 11:09:46 GMT
connection: Keep-Alive
content-length: 0
location: /v2/subscriptions/56fa627aeca5a9f3a56d9e58
```

#### Doc associated in mongoDB
```
{"reference": "http://localhost:1234", "throttling": 0L, "expression": {"q": "", "geometry": "", "georel": "", "coords": ""}, "format": "JSON", "entities": [{"type": "room", "id": ".*", "isPattern": "true"}], "attrs": ["temperture_0"], "_id": ObjectId("56fa627aeca5a9f3a56d9e58"), "conditions": [{"type": "ONCHANGE", "value": ["temperature_0"]}], "expiration": 1459864800L}
```

### PR DESCRIPTION
**Pull request by fgalan**
Fixed #1961

No need of new test harness: behave test will cover this. In addition, no need of CNR entry for an "internal" fix like this, without effect in the functionality.

### CODE REVIEW

File: src/lib/mongoBackend/mongoSubscribeContext.cpp [hunk 1]
```diff
@@ -56,7 +57,7 @@ HttpStatusCode mongoSubscribeContext
   const std::vector<std::string>&      servicePathV
 )
 {
-    std::string        servicePath           = (servicePathV.size() == 0)? "" : servicePathV[0];    
+    std::string        servicePath           = (servicePathV.size() == 0)? DEFAULT_SERVICE_PATH : servicePathV[0];
```
Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubscribeContext.cpp]: Not really part of this PR, just a question ...
We have TWO default service paths,   "/" and "/#", I imagine mongoSubscribeContext uses "/#" as default, as it is some kind of query  ...
However, the define 'DEFAULT_SERVICE_PATH' doesn't really indicate which of the two is in question ...

**Comment by crbrox** [comment]
LGTM

File: src/lib/mongoBackend/mongoSubscribeContext.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubscribeContext.cpp]: Fixed in c24a114

File: doc/manuals/user/service_path.md [hunk 2]
```diff
@@ -54,8 +54,8 @@ Some additional remarks:
     entities created without Fiware-ServicePath (or that don't include
     service path information in the database) belongs to a root scope
     "/" implicitely. All the queries without using Fiware-ServicePath
-    are on "/\#" implicitely. This behavior ensures backward
-    compatibility to pre-0.14.0 versions.
+    (including subscriptions) are on "/\#" implicitely. This behavior
```
Comment (kzangeli) [inline doc/manuals/user/service_path.md]: typo: implicitly

Comment (fgalan) [inline doc/manuals/user/service_path.md]: Fixed in 756c40e5c11b294a098bf6abe1dde346af179a30

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**